src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
+        return NGX_OK;
```

Comment:
@rainingmaster In your case, you should just use `ngx.exit(ngx.ERROR)` to abort the response body. Data already sent out on the wire cannot be undone. The client should handle the case of a truncated response when it detects truncation. Your current way of cheating client by sending a last chunk terminator just makes client impossible to detect truncation.